rest-ehr-scape/src/main/java/org/ehrbase/rest/ehrscape/controller/CompositionController.java

```diff
@@ -81,19 +93,19 @@ public CompositionController(CompositionService compositionService) {
     }
 
 
-    @GetMapping(path = "/{id}")
+    @GetMapping(path = "/{uid}")
     @ApiOperation(value = "Retrieve a Composition by Id")
-    public ResponseEntity<CompositionResponseData> getComposition(@ApiParam(value = "Id of the Composition in the form uuid::version") @PathVariable("id") String compositionId, @ApiParam(value = "Format of the Composition", allowableValues = "XML, ECISFLAT, FLAT, RAW") @RequestParam(value = "format", defaultValue = "XML") CompositionFormat format) {
-        final Integer version;
-        final UUID uuid;
-        if (compositionId.contains("::")) {
-            version = getCompositionVersion(compositionId); //version number is inorder: 1, 2, 3 etc.
-            uuid = getCompositionUid(compositionId);
-        } else {
-            uuid = getCompositionUid(compositionId);
-            version = null;
+    public ResponseEntity<CompositionResponseData> getComposition(@ApiParam(value = "Id of the Composition in the form uuid::version") @PathVariable("uid") String compositionUid,
+                                                                  @ApiParam(value = "Format of the Composition", allowableValues = "XML, ECISFLAT, FLAT, RAW") @RequestParam(value = "format", defaultValue = "XML") CompositionFormat format) {
+
+        UUID identifier = getCompositionIdentifier(compositionUid);
```

Comment:
@subigre is this the openEHR API? because from that, the uid parameter is "version_uid" which is not A UUID, is a OBJECT_VERSION_ID which has the format uuid::string::string, like 8849182c-82ad-4088-a07f-48ead4180515::openEHRSys.example.com::1

https://specifications.openehr.org/releases/ITS-REST/latest/ehr.html#composition-composition-get

Reply:
No the method implements EhrScape REST API that mentions the following (see screenshot above):

> The ID can be specified as a complete unique ID with the domain and version part (8849182c-82ad-4088-a07f-48ead4180515::ehrscape.com::1), or just the identifier part (8849182c-82ad-4088-a07f-48ead4180515), in which case the last version of the composition will be returned.



![image](https://user-images.githubusercontent.com/28752830/134144252-e1327a67-6ff2-465c-b729-511ffa4f2a4c.png)